Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth-2
# 272 Infirm Adding & Scaling Age of Generated Characters

Resolves #272


## Changelog:
- Characters now get infirm event based on their age and race
- Scales the age of generated characters, ie all night elves won't spawn in the age range 16-80 like humans do


## Developer changelog:
- Added age_in_relation_to_humans_more_than_trigger which multiplies the supplied age by a factor depending on target race.
- Changed health.7000 event (infirm event) to trigger depending on the humans age "threshold" multiplied by age_in_relation_to_humans_more_than_trigger.
- Scales the age of generated characters according to their multiplier and the base "human age"


# How to test:
* Check for reasonable ages:
    * Run events that spawns new characters
    * Check generated characters
    * Check rulers and courtiers in startdates.
    * Check events related to age so that they work
* Check marriage score so that there is nothing weird there relating to age
* Check who AI is marrying: Should marry people of roughly the same age, and prefer those of the same race:
    * Are old rulers marrying correctly?
    * Young ones?
    * With heir?
    * Without heir?
* No one having an age higher than 580 is intended for now #292 
* Don't report #285

# TODO:
* Dragons?
- [x] Check for `age`
- [x] Check for `MAX_FEMALE_REPRODUCTION_AGE` and other childbirth defines
- [x] Fixed `divide` errors.
- [x] Check `age = value`.
- [x] Check `marriage_fertile_age_value`
- [x] Check `marriage_fertile_age_reverse_value`
- [x] Check `num_perk_points`
- [x] Check `example_total_family_age`
- [x] Check `racial_age_multiplier_value` being targeting the right person
